pkg/daemon/daemon.go

```diff
@@ -319,9 +319,15 @@ func (dn *Daemon) bootstrapNode() error {
 		return err
 	}
 	dn.node = node
+	if _, err := os.Stat("/etc/defaults/rhcos/reboot-needed"); err == nil {
```

Comment:
IMO, `/etc/defaults` is a legacy of SySV init; no reason to put anything new there.

The right place for this is more likely `/run/machine-config-operator/reboot-needed` or so.  (We also have `/etc/machine-config-daemon` used for `/etc/machine-config-daemon/state.json` but I think `/run` makes more sense since it goes away automatically if the machine is rebooted)

Reply:
cc @ashcrow